src/scenario.cpp

```diff
@@ -165,6 +167,19 @@ void scenario::check_definition() const
     check_traits( _forced_traits, id );
     check_traits( _forbidden_traits, id );
     MapExtras::get_function( _map_special ); // triggers a debug message upon invalid input
+
+    for( auto &m : _missions ) {
+        if( !m.is_valid() ) {
+            debugmsg( "starting mission %s for scenario %s does not exist", m.c_str(), id.c_str() );
+        }
+
+        if( !std::any_of( m->origins.begin(), m->origins.end(), []( mission_origin origin ) {
+        return origin == ORIGIN_GAME_START;
+    } ) ) {
```

Comment:
Lambdas clamped into conditional statements look rather ugly. I'd assign this one to a temporary.

Reply:
Went with the `std::find`--as noted, the lambda is unnecessary here.